flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -2090,7 +2090,7 @@ class ScalarExprLowering {
     }
 
     const Fortran::lower::IntrinsicArgumentLoweringRules *argLowering =
-        Fortran::lower::getIntrinsicArgumentLowering(name);
+        Fortran::lower::getIntrinsicArgumentLowering(name.str());
```

Comment:
Nit: `.str()` leftover from the `getIntrinsicArgumentLowering` change you rolled back.